src/core/include/openvino/pass/pattern/op/optional.hpp

```diff
@@ -4,18 +4,44 @@
 
 #pragma once
 
-// #include "openvino/core/node.hpp"
 #include "openvino/pass/pattern/op/pattern.hpp"
 
 namespace ov {
 namespace pass {
 namespace pattern {
 namespace op {
+
 /// A submatch on the graph value which contains optional op types defined in constructor.
+/// `Optional` pattern supports multi input operations. In this case the pattern checks all pattern input branches
+/// while a branch will be matched with the graph.
 /// The match is succeed in case of full graphs matching or extended by one of optional type graph or pattern.
 /// Otherwise fails.
-/// Important note: graph can include only one optional op in the end of graph vs pattern.
-/// Optional op can contain only 1 in and 1 out
+/// Important notes:
+///   1. a graph can include only one optional op in the end of graph vs pattern.
+///   2. Operation types should have the same input number
+///   3. Pattern can not be created with empty `inputs`
+//
+//  +------+   +------+                       +------+  +------+      +------+     +------+
+//  | op_0 |   | op_1 |                       | op_0 |  | op_1 |      | op_0 |     | op_1 |
+//  +------+   +------+                       +------+  +------+      +------+     +------+
+//     |           |                              |        |              |            |
+//     V           V                              V        V              |            |
+//  +-------------------+                    +---------------------+      |            |
+//  | optional<op_type> |      =======>>>    | wrap_type<op_types> |      |            |
+//  +-------------------+                    +---------------------+      |            |
+//          |                                        |                    |            |
+//          V                                        +------------------+ | +----------+
+//       +------+                                                       | | |
+//       | op_3 |                                                       V V V
+//       +------+                                                    +--------+
+//                                                                   |   Or   |
```

Comment:
It looks like, the pattern will match if only 2nd input of Transpose is matched
e.g. 
<img width="373" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/21181928/4221af60-bff0-4c47-a219-a7f4eb6e61a1">

I believe  it's not what we expected from Optional

Reply:
![image](https://github.com/openvinotoolkit/openvino/assets/22008923/4af1512f-b558-4571-8850-4053fa9e1abb)

Reply:
ok, this example works. 

But I'm not sure that we should match both options:
1. Relu-> relu
2.  Const -> relu 
according to the graph and the code
<img width="111" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/21181928/54aff2a8-ae2d-4274-ba04-1abf22d86ad4">

The Constant (2nd input to Transpose) exists only to create Transpose operation in the pattern/the model
Should it participate in the pattern matching if Transpose is not in the model?

I believe that in most cases we are only interested in the 1st ("data") input of operation ("Relu" in this case).
Otherwise many false positive matches will be triggered. 
This can have a high impact on performance of matching.